plugins/woocommerce/src/Caches/OrderCountCache.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\Caches;
+
+use Automattic\WooCommerce\Caching\CacheException;
+use Automattic\WooCommerce\Caching\ObjectCache;
+use Automattic\WooCommerce\Enums\OrderStatus;
+use Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore;
+use Automattic\WooCommerce\Utilities\OrderUtil;
+
+/**
+ * A class to cache counts for various order statuses.
+ */
+class OrderCountCache extends ObjectCache {
+
+	/**
+	 * Default value for the duration of the objects in the cache, in seconds
+	 * (may not be used depending on the cache engine used WordPress cache implementation).
+	 *
+	 * @var int
+	 */
+	protected $default_expiration = DAY_IN_SECONDS;
```

Comment:
This day expiration is a temporary solution until we get a [better background process in place](https://github.com/woocommerce/woocommerce/issues/54819).

If the orders do become out of sync, this guarantees that it can't remain that way for more than a day (or until cache is flushed).